phper/src/builder/function.rs

```diff
@@ -0,0 +1,101 @@
+use crate::builder::argument::Argument;
+use crate::builder::argument::ArgumentSet;
+use crate::builder::data_type::AtomicReturnType;
+use crate::builder::data_type::AtomicType;
+use crate::builder::describe::Describe;
+use crate::builder::handler::Context;
+use crate::builder::handler::FunctionHandler;
+use crate::builder::handler::State;
+use crate::builder::value::Value;
+use crate::errors::Result;
+
+pub struct Function<F>
+where
+    F: Fn(State, Context, ArgumentSet) -> Result<Value>,
```

Comment:
note to self: should be less restrictive, maybe `Result<T> where T: TryInto<ZVal>` instead of `Result<Value>`?

Reply:
Maybe the former is better.

Reply:
yep, will go back to it later, as i figure how I'm implementing the exception abstraction :)